src/routes/tags.rs

```diff
@@ -302,75 +301,41 @@ pub async fn game_version_delete(
 }
 
 #[derive(serde::Serialize)]
-pub struct LicenseQueryData {
+pub struct License {
     short: String,
     name: String,
 }
 
 #[get("license")]
-pub async fn license_list(
-    pool: web::Data<PgPool>,
-) -> Result<HttpResponse, ApiError> {
-    let results: Vec<LicenseQueryData> = License::list(&**pool)
-        .await?
-        .into_iter()
-        .map(|x| LicenseQueryData {
-            short: x.short,
-            name: x.name,
-        })
-        .collect();
-    Ok(HttpResponse::Ok().json(results))
-}
-
-#[derive(serde::Deserialize)]
-pub struct LicenseData {
-    name: String,
-}
-
-#[put("license/{name}")]
-pub async fn license_create(
-    req: HttpRequest,
-    pool: web::Data<PgPool>,
-    license: web::Path<(String,)>,
-    license_data: web::Json<LicenseData>,
-) -> Result<HttpResponse, ApiError> {
-    check_is_admin_from_headers(req.headers(), &**pool).await?;
-
-    let short = license.into_inner().0;
-
-    let _id = License::builder()
-        .short(&short)?
-        .name(&license_data.name)?
-        .insert(&**pool)
-        .await?;
+pub async fn license_list() -> HttpResponse {
+    let licenses = spdx::identifiers::LICENSES;
+    let mut results: Vec<License> = Vec::with_capacity(licenses.len());
+
+    for (short, name, _) in licenses {
+        results.push(License {
+            short: short.to_string(),
+            name: name.to_string(),
+        });
+    }
 
-    Ok(HttpResponse::NoContent().body(""))
+    HttpResponse::Ok().json(results)
 }
 
-#[delete("license/{name}")]
-pub async fn license_delete(
-    req: HttpRequest,
-    pool: web::Data<PgPool>,
-    license: web::Path<(String,)>,
+#[get("license/{id}")]
+pub async fn license_text(
+    params: web::Path<(String,)>,
 ) -> Result<HttpResponse, ApiError> {
-    check_is_admin_from_headers(req.headers(), &**pool).await?;
-
-    let name = license.into_inner().0;
-    let mut transaction =
-        pool.begin().await.map_err(models::DatabaseError::from)?;
+    let license_id = params.into_inner().0;
 
-    let result = License::remove(&name, &mut transaction).await?;
-
-    transaction
-        .commit()
-        .await
-        .map_err(models::DatabaseError::from)?;
-
-    if result.is_some() {
-        Ok(HttpResponse::NoContent().body(""))
-    } else {
-        Ok(HttpResponse::NotFound().body(""))
+    if let Some(license) = spdx::license_id(&*license_id) {
+        return Ok(HttpResponse::Ok()
+            .content_type("text/plain")
+            .body(license.text()));
     }
+
+    Err(ApiError::InvalidInput(
+        "Invalid SPDX identifier specified".to_string(),
+    ))
 }
```

Comment:
oops, I should've said in discord but for consistency this should probs be in JSON (just one field `body`)